### ISSUE TITLE
Unable to create PKGBUILD file for pacman

### ISSUE THREAD

**Issue by WorldSEnder** [hypro/hypro#2]
I'm working on arch linux and tried to write a PKGBUILD file to organize installation of HyPro. Yet, during building, I get a ton of error in the likes of `undefined reference to '__gmpq_clear'` when the tests are being built. I use the following PCKBUILD file which I thought to be a straight forward implementation of your build instructions:

```
pkgname=HyPro
pkgdesc="A C++ state set representation library for the analysis of hybrid systems"
url="https://github.com/hypro/hypro"
arch=("i686" "x86_64")
license=("mit")
depends=("gmp" "boost" "glpk")
makedepends=("make" "cmake")
conflicts=()
replaces=()
backup=()
source=("$pkgname::git+https://github.com/hypro/hypro.git")
pkgver=r2243.d23623d5
prepare() {
  cd "$pkgname" 
}
pkgver() {
  cd "$pkgname"
  ( set -o pipefail
    git describe --long 2>/dev/null | sed 's/\([^-]*-g\)/r\1/;s/-/./g' ||
    printf "r%s.%s" "$(git rev-list --count HEAD)" "$(git rev-parse --short HEAD)"
  )
}
pkgrel=1
build() {
  cd "$pkgname"
  cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr .
  make resources
  make
}
check() {
  cd "$pkgname"
  make test
}
package() {
  cd "$pkgname"
  make DESTDIR="$pkgdir/" install
}
md5sums=('SKIP')
```

And I run try to install via `makepkg`. Note that when I redo the installation without cleaning up `makepkg -f`, then it will succeed compiling.

**Comment by modass** [comment]
I am not really familiar with pkgconfig, but I noticed the dependencies metion `gmp`, whereas we defenitely need the c++ interface `gmpxx`. Does that solve your issue? Other than that - does the above specification build out of source?

**Comment by WorldSEnder** [comment]
The package containing `gmpxx` is `gmp` so that should be fine. The script builds in-source, but I have modified it to build out of source since I posted. I will make a gist with the files to track changes and revision

**Comment by WorldSEnder** [comment]
For completeness I should mention that I have CaRL already installed on the system, yet when I build HyPro, it still downloads and builds it itself. That could influence the behaviour.

**Comment by WorldSEnder** [comment]
CaRL has a community package for arch here: https://aur.archlinux.org/packages/carl/

**Comment by modass** [comment]
Yes - I see the conversation with nafur as well. I think the problem why hypro downloads carl again is that so far we've used the global cmake registering of carl to find its location. I will update the find script for carl and try on a virtual machine to get that running.

**Comment by modass** [comment]
Okay, I think I found the issue and fixed it. Can you please verify/confirm this?

**Comment by WorldSEnder** [comment]
Should be solved together with #6 in #8
